x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/evaluation/classification/AucRoc.java

```diff
@@ -182,42 +182,43 @@ public void process(Aggregations aggs) {
         Filter classAgg = aggs.get(TRUE_AGG_NAME);
         Nested classNested = classAgg.getAggregations().get(NESTED_AGG_NAME);
         Filter classNestedFilter = classNested.getAggregations().get(NESTED_FILTER_AGG_NAME);
+
+        Filter restAgg = aggs.get(NON_TRUE_AGG_NAME);
+        Nested restNested = restAgg.getAggregations().get(NESTED_AGG_NAME);
+        Filter restNestedFilter = restNested.getAggregations().get(NESTED_FILTER_AGG_NAME);
+
+        long filteredDocCount = classNestedFilter.getDocCount() + restNestedFilter.getDocCount();
+        long totalDocCount = classAgg.getDocCount() + restAgg.getDocCount();
+
         if (classAgg.getDocCount() == 0) {
             throw ExceptionsHelper.badRequestException(
                 "[{}] requires at least one [{}] to have the value [{}]",
                 getName(), fields.get().getActualField(), className);
         }
-        if (classNestedFilter.getDocCount() == 0) {
+        if (classNestedFilter.getDocCount() < classAgg.getDocCount()) {
             throw ExceptionsHelper.badRequestException(
-                "[{}] requires at least one [{}] to have the value [{}]",
-                getName(), fields.get().getPredictedClassField(), className);
+                "[{}] requires that [{}] appears as one of the [{}] for every document (appeared for {} out of {})",
```

Comment:
Do you think it would be good to give the potential solution for the analytics case? Maybe something like "This is probably caused by the `num_top_classes` value being less than the total number of classes in the data frame analytics training configuration"

I am sure there is better wording :).